spl/src/token_2022.rs

```diff
@@ -382,7 +377,10 @@ pub fn get_account_data_size<'info>(
         .and_then(|(key, data)| {
             if key != *ctx.program.key {
                 Err(anchor_lang::solana_program::program_error::ProgramError::IncorrectProgramId)
-            } else {
+            } else if data.len() != 8 {
```

Comment:
Why did you have a check on the data len? The check is already performed just after